.gitignore

```diff
@@ -1,4 +1,31 @@
-#################
+# Created by https://www.toptal.com/developers/gitignore/api/meson
+# Edit at https://www.toptal.com/developers/gitignore?templates=meson
+
+### Meson ###
+# subproject directories
+/subprojects/*
+!/subprojects/*.wrap
+
+# Meson Directories
+meson-logs
+meson-private
+
+# Meson Files
+meson_benchmark_setup.dat
+meson_test_setup.dat
+sanitycheckcpp.cc     # C++ specific
+sanitycheckcpp.exe    # C++ specific
+
+# Ninja
+build.ninja
+.ninja_deps
+.ninja_logs
+
+# Misc
+compile_commands.json
+
+# End of https://www.toptal.com/developers/gitignore/api/meson
+# #################
```

Comment:
This almost entirely looks buggy, so I don't know what toptal.com's rationale here is.

`/subprojects/*` is reasonable enough -- meson subprojects will be downloaded there so you only want to check in `.wrap` files. Appropriately, it is rooted in `/`.

The rest are all files that are created inside of the build directory. Excluding any one of them -- or even a bunch of them -- is missing the point that it fails to gitignore every possible `*.o` file, or your `*.so` / `*.dll` outputs, or many many other things generated as part of a build.

... it's also redundant due to https://mesonbuild.com/FAQ.html#how-do-i-ignore-the-build-directory-in-my-vcs since this is one of the Quality of Life advantages of using meson; you do not need to gitignore the build directory at all.

That maybe should be fixed somewhere in this generator tool.